src/main/groovy/com/twcable/grabbit/jcr/JcrNodeDecorator.groovy

```diff
@@ -136,6 +136,37 @@ class JcrNodeDecorator {
     }
 
     /**
+     * Finds out nearest versionable ancestor for a node and performs a checkout
+    */
+    void checkoutNode() {
+        try {
+            JcrNodeDecorator decoratedVersionableAncestor = findVersionableAncestor()
+            if (decoratedVersionableAncestor && !decoratedVersionableAncestor.isCheckedOut()) {
+                decoratedVersionableAncestor.session.workspace.versionManager.checkout(decoratedVersionableAncestor.path)
+            }
+        }
+        catch (Exception exception) {
```

Comment:
It could throw UnsupportedRepositoryOperationException, LockException or RepositoryException.. the log statement includes the exception message... IMO if we are not doing any different for different types of exceptions, having single exception is cleaner...